utils/fdb_wrapper_cpp/src/fdbcs.cpp

```diff
@@ -206,6 +223,312 @@ std::shared_ptr<FDBDataBase> DataBaseCreator::createDataBase(const std::string c
+                            const ByteArray& blob)
+{
+  const size_t blobSizeInBytes = blob.size();
+  if (!blobSizeInBytes)
+    return commitKey(dataBase, key, "");
+
+  const uint32_t numDataBlocks = (blobSizeInBytes + (dataBlockSizeInBytes_ - 1)) / dataBlockSizeInBytes_;
+  const uint32_t treeLen = std::ceil(log(numKeysInBlock_, numDataBlocks));
+  Keys currentKeys{key};
+  auto numKeyLevelMap = computeNumKeysForEachTreeLevel(treeLen, numDataBlocks);
+
+  std::unordered_map<Key, Block> map;
+  map[key] = {0, std::string()};
+  Keys keysInTnx;
+  size_t currentTnxSize = 0;
+
+  for (uint32_t currentLevel = 0; currentLevel < treeLen; ++currentLevel)
+  {
+    const uint32_t nextLevelKeyNum = numKeyLevelMap[currentLevel];
+    auto nextLevelKeys = generateKeys(nextLevelKeyNum);
+    uint32_t nextKeysIt = 0;
+    for (uint32_t i = 0, size = currentKeys.size(); i < size && nextKeysIt < nextLevelKeyNum; ++i)
+    {
+      const auto& currentKey = currentKeys[i];
+      auto& block = map[currentKey];
+      for (uint32_t j = 0; j < numKeysInBlock_ && nextKeysIt < nextLevelKeyNum; ++j, ++nextKeysIt)
+      {
+        const auto& nextKey = nextLevelKeys[nextKeysIt];
+        insertKey(block, nextKey);
+        map[nextKey] = {0, std::string()};
+      }
+      if (currentTnxSize + (keySizeInBytes_ + block.second.size()) >= maxTnxSize_)
+      {
+        RETURN_ON_ERROR(commitKeys(dataBase, map, keysInTnx));
+        currentTnxSize = 0;
+        keysInTnx.clear();
+      }
+      currentTnxSize += block.second.size() + keySizeInBytes_;
+      keysInTnx.push_back(currentKey);
+    }
+    currentKeys = std::move(nextLevelKeys);
+  }
+
+  size_t offset = 0;
+  for (uint32_t i = 0; i < numDataBlocks; ++i)
+  {
+    const auto& currentKey = currentKeys[i];
+    auto& block = map[currentKey];
+    offset = insertData(block, blob, offset);
+    if (currentTnxSize + (keySizeInBytes_ + block.second.size()) >= maxTnxSize_)
+    {
+      RETURN_ON_ERROR(commitKeys(dataBase, map, keysInTnx));
+      currentTnxSize = 0;
+      keysInTnx.clear();
+    }
+    keysInTnx.push_back(currentKey);
+    currentTnxSize += block.second.size() + keySizeInBytes_;
+  }
+
+  if (currentTnxSize)
+    RETURN_ON_ERROR(commitKeys(dataBase, map, keysInTnx));
+
+  return true;
+}
+
+std::pair<bool, Keys> BlobHandler::getKeysFromBlock(const Block& block)
+{
+  Keys keys;
+  const auto& blockData = block.second;
+  if (blockData.size() > blockSizeInBytes_)
+    return {false, {""}};
+
+  uint32_t offset = 1;
+  for (uint32_t i = 0; i < numKeysInBlock_ && offset + keySizeInBytes_ <= blockData.size(); ++i)
+  {
+    Key key(blockData.begin() + offset, blockData.begin() + offset + keySizeInBytes_);
+    keys.push_back(std::move(key));
+    offset += keySizeInBytes_;
+  }
+
+  return {true, keys};
+}
+
+bool BlobHandler::isDataBlock(const Block& block)
+{
+  return block.second.compare(0, keyBlockIdentifier.size(), keyBlockIdentifier) != 0;
```

Comment:
Can you ellaborate on the semanitcs of this expression @denis0x0D?

Reply:
We have to identify the data block and the key block, every block has special header starting from the beginning of the block, so this function returns true when we processing the data block and false otherwise.